### PR TITLE
add watchpoints to mipsy_web

### LINKED ISSUE #250: watchpoints in mipsy_web
now that watchpoints are a part of `mipsy_lib`, having mipsy_web be able to break on the reading/changing of a register would be useful. Adding buttons for watchpoints in the used/all registers tab would probably be most intuitive

### PR DESCRIPTION
**Pull request by joooooooooooooooooooooooooooooooooooosh**
fixes #250

### CODE REVIEW

**Comment by SpanishPear** [comment]
does this work now btw?

**Comment by joooooooooooooooooooooooooooooooooooosh** [comment]
> does this work now btw?

It should yeah

**Comment by SpanishPear** [comment]
Sorry, not sure why I didn't get a notif.. will look :)

File: crates/mipsy_web/README.md [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 # About
```
Comment (SpanishPear) [inline crates/mipsy_web/README.md:11]: thanks for the updates hehe

File: crates/mipsy_web/src/components/decompiled.rs [hunk 2]
```diff
@@ -138,7 +139,7 @@ pub fn render_decompiled(props: &DecompiledProps) -> Html {
 }
 
 #[function_component(StopIconOutline)]
-fn stop_icon_outline() -> Html {
+pub(crate) fn stop_icon_outline() -> Html {
```
Comment (SpanishPear) [inline crates/mipsy_web/src/components/decompiled.rs:142]: should just be `pub` but should be fine

File: crates/mipsy_web/src/state/update.rs [hunk 3]
```diff
@@ -88,7 +88,7 @@ pub fn handle_response_from_worker(
                 }))
             }
 
-            _ => unreachable!("cannot update binary if there is no binary"),
```
Comment (SpanishPear) [inline crates/mipsy_web/src/state/update.rs:91]: why?

**Review by SpanishPear — APPROVED** [review]
happy for you to merge :D works good upon testing

small note is that it's not immediately clear that the buttons for read/write watchpoints are, that vs like "icons" for if the register has been read/written to. But, I think not a big deal. 

Would also be good to see a setting to toggle them off (as, most students wont use unless seriously stuck) - but effort.

File: crates/mipsy_web/src/state/update.rs (hunk 3 above)
Comment (joooooooooooooooooooooooooooooooooooosh) [reply crates/mipsy_web/src/state/update.rs:91]: I don't remember the circumstances now but I ran into it somehow during testing

**Pull request merged**